hhnk_threedi_tools/core/checks/sqlite/sqlite_main.py

```diff
@@ -494,82 +494,55 @@ def create_grid_from_sqlite(self, sqlite_path, dem_path, output_folder):
                 output_path=os.path.join(output_folder, f"{i}.gpkg"),
             )
 
-    def run_cross_section(self):
-        """TODO add docstring en implementeren in plugin"""
+    def run_cross_section(self, database):
+        """Deze functie een loop maken over het cross sections om een of meer intersect punten (cross_sections) vinden
+           Het ga een foutmelding geven als 1 of meer punten vinden"""
         try:
-            cross_section_point = self.model.read_table(table_name="v2_cross_section_location",  
-                                  columns=["id", "channel_id", "reference_level", "bank_level", "geometry"])
-            cross_section_point.rename({"id": "cross_loc_id"}, axis=1, inplace=True)
-
-            cross_section_buffer_gdf = cross_section_point
-            cross_section_buffer_gdf["geometry"] = cross_section_buffer_gdf.buffer(0.5) 
-            
-            cross_section_join =  gpd.sjoin(cross_section_buffer_gdf, cross_section_point,
-                              how="inner", predicate="intersects")
-            cross_section_join['new'] = np.where((cross_section_join["channel_id_right"]==cross_section_join["channel_id_left"]), cross_section_join['cross_loc_id_left'], np.nan)
-            cross_section_loc_id = cross_section_join.set_index('new')
-            cross_sec_id = [] 
-            counter = []
-            cross_location_id = []
+           
+            cross_section_point = database.execute_sql_selection(query= cross_section_location_query)
+        
+            # gebruik het functie _get_intersected_ points om de snijpunten te krijgen. 
+            intersected_points = _get_intersected_points(cross_section_point)
             
-            for id in cross_section_loc_id.index:
-                # print(id)
-                cross_sec_id.append(id)
-            for _ in cross_sec_id:    
-                cross_location_id.append(_)
-                count = cross_sec_id.count(_)
-                counter.append(count)
-
-            gdf_data = gpd.GeoDataFrame({"cross_loc_id_left":cross_location_id, 'count': counter})
-            gdf_data =  gdf_data.drop_duplicates(['cross_loc_id_left','count'],keep = 'last')
-            gdf_data = gdf_data[gdf_data['count']> 1 ]
-            list_id = gdf_data.cross_loc_id_left.values.tolist()
-            cross_section_warning = cross_section_point[cross_section_point['cross_loc_id'].isin(list_id)]
-            return cross_section_warning 
+            # Een foutmelding geven als de volgen de conditie.  
```

Comment:
Printing a full table isnt great on the main

Reply:
deleted printing lines  from both functions:  run_cross_section_vertex and run_cross_section